BuildingBlocks/src/UnitTestTools/Extensions/DateTimeAssertionsExtensions.cs

```diff
@@ -11,20 +12,28 @@ public static AndConstraint<DateTimeAssertions> Be(this DateTimeAssertions it, s
 
     public static AndConstraint<NullableDateTimeAssertions> BeEndOfHour(this NullableDateTimeAssertions it)
     {
-        return it.NotBeNull().
-            And.HaveMinute(59).
-            And.HaveSecond(59);
+        return it.NotBeNull()
+            .And.HaveYear(it.Subject!.Value.Year)
+            .And.HaveMonth(it.Subject!.Value.Year)
+            .And.HaveDay(it.Subject!.Value.Year)
+            .And.HaveHour(it.Subject!.Value.Hour)
```

Comment:
Aren't all of these garanteed to be true? Also, the `HaveMonth` and `HaveDay` methods seem to be being called with the wrong parameter.

Reply:
> Aren't all of these garanteed to be true?
You are right, of course. 🤣 I have no idea what I was expecting to achieve.

> Also, the HaveMonth and HaveDay methods seem to be being called with the wrong parameter.
Yup. Copy&paste error...

I will fix it.